activerecord/test/cases/relation/where_clause_test.rb

```diff
@@ -181,6 +181,39 @@ class WhereClauseTest < ActiveRecord::TestCase
       assert_equal WhereClause.empty, WhereClause.empty.or(where_clause)
     end
 
+    test "or places common conditions before the OR" do
+      wcs = (0..6).map do |i|
+        WhereClause.new([table["col_#{i}"].eq(bind_param(i))])
```

Comment:
Just tried, that doesn't work because WhereClause#or isn't call for both sides, and WhereClause#or always adds a And node around the predicates, even if there only one.

I'll tweak the test.